astropy/utils/tests/test_data.py

```diff
@@ -340,6 +342,7 @@ def test_download_with_sources_and_bogus_original(
         assert is_url_in_cache(u)
 
 
+@pytest.mark.skip(reason="causes mystery segfault! possibly bug #10008")
```

Comment:
Maybe we do a `skipif` so this runs for Python 3.8+? Same comment for all the other similar skips.

Reply:
Python 3.6 seems to be fine too.

The point of these tests was to ensure that the locking code got registered by the coverage tools - there are multiprocessing-based tests that explore a variety of concurrent paths through the locking code but the subprocesses don't show on the coverage, so I ran some thread-based tests instead to check all my branches were covered. So with the thread-based tests disabled the coverage tools will see less, even apart from the code that's skipped. Conditional skipping should help a little.